internal/common/metrics.go

```diff
@@ -49,25 +49,36 @@ func sendCollectDMetrics(metricPerMinute int, duration time.Duration) error {
 
 	defer client.Close()
 
-	ticker := time.NewTicker(60 * time.Second)
+	ticker := time.NewTicker(sendingInterval)
 	defer ticker.Stop()
 	endTimeout := time.After(duration)
 
 	for {
 		select {
 		case <-ticker.C:
-			for t := 0; t < metricPerMinute; t++ {
-				err = client.Write(ctx, &api.ValueList{
+			for t := 1; t <= metricPerMinute/2; t++ {
+				_ = client.Write(ctx, &api.ValueList{
```

Comment:
why are we ignoring this err now

Reply:
Since the [second write](https://github.com/aws/amazon-cloudwatch-agent-test/pull/178/files/583f794c60c318695351896820d9ec205933630e#diff-781bb3810ede78771b4deb63bb6bb797a2d06434f0d844429b5ad16a20304e0fR71-R81) will handle the error and if its failed the second time, then it would failed the first time. In case such as the [buffer being filled up](https://github.com/aws/amazon-cloudwatch-agent-test/pull/178/files/583f794c60c318695351896820d9ec205933630e#diff-781bb3810ede78771b4deb63bb6bb797a2d06434f0d844429b5ad16a20304e0fR82-R84), the second one would be the culprit for filling the disk up so we don't need to validate the first write.